File: templates/renga_net_template/init_app.cs
using System;
using Renga;

namespace renga_net_template
{
    public class init_app : Renga.IPlugin
    {
        Renga.ActionEventSource follow_action;
        public bool Initialize(string pluginFolder)
        {
            Renga.Application renga_app = new Renga.Application();
            Renga.IUI renga_ui = renga_app.UI;
            Renga.IUIPanelExtension panel = renga_ui.CreateUIPanelExtension();

            Renga.IAction our_button = renga_ui.CreateAction();
            our_button.ToolTip = "Test plugin";
            our_button.DisplayName = "Start test message box";

            Renga.IImage icon = renga_ui.CreateImage();
            icon.LoadFromFile(pluginFolder + "\\logo.png");
            our_button.Icon = icon;

            follow_action = new ActionEventSource(our_button);
            follow_action.Triggered += (sender, args) =>
            {
                renga_ui.ShowMessageBox(MessageIcon.MessageIcon_Info, "Сообщение", "Привет, Renga!");
            };

            panel.AddToolButton(our_button);
            renga_ui.AddExtensionToPrimaryPanel(panel);

            return true;
        }
        public void Stop()
        {
            follow_action.Dispose();
        }
    }
}

File: lessons/blok4/nwc_export3/nwc_export3/init_app.cpp
#include "pch.h"
#include "init_app.hpp"
#include "string"
#include "export_nwc.hpp"
#include "Renga\CreateApplication.hpp"
class button_run :public Renga::ActionEventHandler {
public:
	button_run(Renga::IActionPtr action) : Renga::ActionEventHandler(action) {}
	void OnTriggered() override
	{
		new export_nwc(renga_application);
		renga_application->UI->ShowMessageBox(Renga::MessageIcon_Info, "Сообщение", "Экспорт завершен!");
	}
	void OnToggled(bool checked) override {}

};
void init_app::addHandler(Renga::ActionEventHandler* pHandler) 
{
	m_handlerContainer.emplace_back(HandlerPtr(pHandler));
}
bool init_app::initialize(const wchar_t* pluginPath)
{
	auto renga_app = Renga::CreateApplication();
	if (renga_app)
	{
		renga_application = renga_app;
		Renga::IUIPtr renga_ui = renga_app->GetUI();
		Renga::IUIPanelExtensionPtr panel = renga_ui->CreateUIPanelExtension();

		Renga::IActionPtr our_button = renga_ui->CreateAction();
		our_button->ToolTip = "Начать экспорт";

		Renga::IImagePtr icon = renga_ui->CreateImage();
		std::wstring plugin_dir(pluginPath);
		icon->LoadFromFile((plugin_dir + L"\\logo.png").c_str());
		our_button->PutIcon(icon);

		this->addHandler(new button_run(our_button));
		panel->AddToolButton(our_button);
		renga_ui->AddExtensionToPrimaryPanel(panel);

	}
	else return false;
	return true;
}
void init_app::stop() {
	this->m_handlerContainer.clear();
}